core/src/main/java/kin/ecosystem/core/data/offer/OfferRepository.java

```diff
@@ -120,11 +121,16 @@ public void removeNativeOfferClickedObserver(@NonNull Observer<NativeSpendOffer>
     @Override
     public boolean addNativeOffer(@NonNull NativeOffer nativeOffer) {
 		Offer offer = nativeOfferList.getOfferByID(nativeOffer.getId());
-		return offer == null && nativeOfferList.addAtIndex(0, nativeOffer);
+		if(offer == null) {
+            offer = Converter.toOffer(nativeOffer);
+            return offer != null && nativeOfferList.addAtIndex(0, offer);
+        }
+		return false;
 	}
 
     @Override
     public boolean removeNativeOffer(@NonNull NativeOffer nativeOffer) {
-        return nativeOfferList.remove(nativeOffer);
+        Offer offer = Converter.toOffer(nativeOffer);
```

Comment:
why there is a logic change?

Reply:
because NativeOffer does not extends Offer anymore.